app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -269,20 +263,25 @@ class NewMessageFragment : Fragment() {
         }
     }
 
-    private fun initDraftAndViewModel() {
-
-        newMessageViewModel.initDraftAndViewModel(
-            intent = requireActivity().intent,
-            navArgs = newMessageActivityArgs,
-        ).observe(viewLifecycleOwner) { isSuccess ->
-            if (isSuccess) {
-                showKeyboardInCorrectView()
-            } else {
-                requireActivity().apply {
-                    showToast(R.string.failToOpenDraft)
-                    finish()
-                }
+    private fun initDraftAndViewModel() = with(newMessageViewModel) {
+
+        if (initResult.value == null) initDraftAndViewModel(intent = requireActivity().intent)
+
+        initResult.observeNotNull(viewLifecycleOwner) { (isSuccess, signatures) ->
+
+            if (!isSuccess) with(requireActivity()) {
+                showToast(R.string.failToOpenDraft)
+                finish()
+                return@observeNotNull
             }
+
+            showKeyboardInCorrectView()
```

Comment:
Focus is not set correctly anymore in the 4 different cases (REPLY, REPLY_ALL, FORWARD, NEW_MAIL).

I figured this out because according to your new code, the method to give focus is now called everytime the activity is recreated which would be wrong anyway, because we only want to give focus the very first time the fragment is created